### ISSUE TITLE
[NEW FEATURE] Run once a day automatically

### ISSUE THREAD

**Issue by commonslibs** [charlesbel/Microsoft-Rewards-Farmer#235]
I've made another change to the original code so that we can have the process started on a machine and run once a day starting at one hour.

The time of day at which the process should be executed and whether or not it has to be executed this way and not just once would be informed through the parameter at startup, so if this parameter is not passed it would be executed only once .

It is important to note that I have tried to put the process in a linux "cron" and it does not run correctly when started from "cron" so that is why I have included this functionality so that it is python that executes the process with the periodicity that we want.

Do you find it interesting that it includes another pull request with this new functionality?

Greetings

**Comment by ecsstasyyy** [comment]
can u send me the new code please i need it, it keeps on being stuck at "logged in" for me

**Comment by commonslibs** [comment]
I'll be on vacation for a few days. When I come back I'll include the code, although it's really very easy what you have to add. I put here briefly the most important code points.

import schedule
.......

if __name__ == "__main__":
    
    # Execute "mainProcess" everyday at 13:00
    schedule.every().day.at("13:00").do(mainProcess)

    # Loop
    while True:
        # Check if there are tasks pending to be executed
        schedule.run_pending()
        time.sleep(1)

Greetings

**Comment by mgrimace** [comment]
I use this as my crontab entry:
`0 6,8 * * * /usr/bin/python3 /home/msbot/Microsoft-Rewards-Farmer/main.py`

I run it twice daily, once at 6 am an again at 8 am as a backup to catch anything it missed the first time, for example finishing searches, etc.

That being said, would you consider using your code + an `-everyday` run argument to enable it manually? That might work for folks who prefer to do this without cron and leave the script running.

I use this on a headless ubuntu server, and would prefer to have the automatic scheduling via a service, cron, or ideally a docker container personally.

**Comment by richplant** [comment]
The following batch file works nicely with Windows Scheduler to run the script as a scheduled task, assuming you're using a venv in the repo path. Just set up a new task to run daily, with the Action definition as follows: 

Start a Program: `cmd` 
Add arguments: `/c <filename>.bat` 
Start in: `<repo_path>`

```
set ORIG_PATH=%CD%
set VENV_FOLDER=<venv_name>
set VENV_PATH=%ORIG_PATH%\%VENV_FOLDER%

call %VENV_PATH%\Scripts\activate.bat

python main.py

call %VENV_PATH%\Scripts\deactivate.bat

exit /B 0

**Comment by commonslibs** [comment]
> I use this as my crontab entry: `0 6,8 * * * /usr/bin/python3 /home/msbot/Microsoft-Rewards-Farmer/main.py`
> 
> the entry follows the order: `cron schedule python location script location`
> 
> tips: use [cronguru](https://crontab.guru) to create your preferred schedule. I run it twice daily, once at 6 am an again at 8 am as a backup to catch anything it missed the first time, for example finishing searches, etc. Note the location to `main.py` will be different depending where you have it installed.
> 
> That being said, would you consider using your code + an `-everyday` run argument to enable it manually? That might work for folks who prefer to do this without cron and leave the script running, e.g., `python3 main.py -everyday`
> 
> I use the script on a headless ubuntu server, and would prefer to have the automatic scheduling via a service, cron, or ideally a docker container.

OK, I'll try "cron" again, although in previous versions of the "Farm" project it didn't work well for me on my ubuntu. If it works well I report it and we could leave this "new feature" out. Thank you so much

**Comment by mgrimace** [comment]
> OK, I'll try "cron" again, although in previous versions of the "Farm" project it didn't work well for me on my ubuntu. If it works well I report it and we could leave this "new feature" out. Thank you so much

No problem and I hope it works for you! Cron never worked well for me either with any other farm projects because, for whatever reason, the script would never properly read the account .json or log files when it ran via cron. This project seems to work for me though via cron and I'm not actually sure why.

**Comment by bobdabear** [comment]
If someone wants to write the script to create this file (with the edit for proper path in main()) then add the cronjob for the created script I can make a pull request to add it to the project or it'd at least it be simple for charlesbel to add so the feature can be added quicker

**Comment by commonslibs** [comment]
I have tested the execution of the script with "cron" in ubuntu and it works correctly. I think the best solution is to run it with cron.

The steps and the modification in the file have been the following:

From command line:
"crontab -e"

Once the file is open, include at the end something similar to the following:

"10 9,11,13 * * * cd /home/user/msrewards && /usr/bin/python3 main.py"

Save the file with ctrl+a and ctrl+x
The included command would run the program three times, once at 9:10 a.m., once at 11:10 a.m., and once at 1:10 p.m.

If deemed appropriate, the "new feature" can be closed

All the best

**Comment by mgrimace** [comment]
> I have tested the execution of the script with "cron" in ubuntu and it works correctly. I think the best solution is to run it with cron.
> 
> The steps and the modification in the file have been the following:
> 
> From command line: "crontab -e"
> 
> Once the file is open, include at the end something similar to the following:
> 
> "10 9,11,13 * * * cd /home/user/msrewards && /usr/bin/python3 main.py"
> 
> Save the file with ctrl+a and ctrl+x The included command would run the program three times, once at 9:10 a.m., once at 11:10 a.m., and once at 1:10 p.m.
> 
> If deemed appropriate, the "new feature" can be closed
> 
> All the best

Thanks for following up with this and the updated cron, I had never thought of using `cd`, and `&&` to get to the directory first which should solve some of the logging and potential login issues.

I'd suggest, for what its worth, to create a PR for the readme to add these instructions so they aren't buried here, I'm sure it'd be very helpful for anyone coming in.

my only suggestion to the wording would be to use the repo name as the folder and possibly generic instructions to maximize accessibility (though my suggestions may be too-much-info, so you can ignore too), e.g.,

> From command line: "crontab -e"
> 
> Once the file is open, include at the end something similar to the following:

`1 9,11,13 * * * cd /home/user/Microsoft-Rewards-Farmer && /usr/bin/python3 main.py` 

> This example command would run the program three times, once at 9:10 a.m., once at 11:10 a.m., and once at 1:10 p.m.

The example follows the order: `[cron schedule] [absolute path to Microsoft-Rewards-Famer folder] && [absolute path to python] main.py`

- cron: use [cronguru](https://crontab.guru/) to create your preferred schedule.
- Python: use `which python` to find the python directory, it is typically `/usr/bin/python3`

> Save the file with ctrl+a and ctrl+x 

Thanks again!
